src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+  }
+
+  /* Initialize memory for arrays */
+  _total_weights = new FP_PRECISION*[_num_azim/2];
+  _sin_thetas = new FP_PRECISION*[_num_azim/2];
+  for (int a=0; a < _num_azim/2; a++) {
+    _total_weights[a] = new FP_PRECISION[_num_polar];
+    _sin_thetas[a] = new FP_PRECISION[_num_polar];
+  }
+
+  /* Compute multiples of sine thetas and weights */
+  for (int a=0; a < _num_azim/4; a++) {
+    for (int p=0; p < _num_polar/2; p++) {
+      _sin_thetas[a][p] = sin(_thetas[a][p]);
+      FP_PRECISION weight = 4.0 * M_PI * _azim_weights[a]
+                            * _azim_spacings[a] * _polar_weights[a][p];
+      weight *= 2.0 * _sin_thetas[a][p];
+      setPolarValues(_total_weights, a, p, weight);
+    }
+  }
+}
+
+
+/**
+ * @brief Converts this Quadrature to a character array of its attributes.
+ * @details The character array includes the number of polar angles, the
+ *          the values of the sine and weight of each polar angle, and the
+ *          product of the sine and weight of each polar angle.
+ * @return a character array of the Quadrature's attributes
+ */
+std::string Quadrature::toString() {
+
+  std::stringstream string;
+
+  string << "Quadrature";
+  string << "\n\t# azim angles  = " << _num_azim;
+  string << "\n\t# polar angles = " << _num_polar;
+
+  string << "\n\tphis = ";
+  if (_phis != NULL) {
+    for (int a = 0; a < _num_azim/4; a++)
+      string << _phis[a] << ", ";
+  }
+
+  string << "\n\tazim weights = ";
+  if (_azim_weights != NULL) {
+    for (int a = 0; a < _num_azim/4; a++)
+      string << _azim_weights[a] << ", ";
+  }
+
+  string << "\n\tthetas = ";
+  if (_thetas != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _thetas[a][p] << ", ";
+    }
+  }
+
+  string << "\n\tpolar weights = ";
+  if (_polar_weights != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _polar_weights[a][p] << ", ";
+    }
+  }
+
+  string << "\n\tsin thetas = ";
+  if (_sin_thetas != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _sin_thetas[a][p] << ", ";
+    }
+  }
+
+  string << "\n\ttotal weights = ";
+  if (_total_weights != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _total_weights[a][p] << ", ";
+    }
+  }
+
+  return string.str();
+}
+
+
+quadratureType Quadrature::getQuadratureType() {
+  return _quad_type;
+}
+
+
+/**
+ * @brief Dummy constructor calls the parent constructor.
+ */
+TYPolarQuad::TYPolarQuad(): Quadrature() {
+  _quad_type = TABUCHI_YAMAMOTO;
+}
+
+
+/**
+ * @brief Set the number of polar angles to initialize.
+ * @param num_polar the number of polar angles (maximum 3)
```

Comment:
The maximum of 3 in the comment is inconsistent with the actual permissible maximum of 6 in the method.